Remove deprecated caffe2::Tensor APIs

jerryzh168 opened the pull request:
Summary: Plan is to remove the APIs we want to deprecate one by one and make sure it still builds in sandcastle and ossci

Reviewed By: ezyang

Differential Revision: D12812029

bddppq reviewed (commented):
Need to update .hip files (they are the .cu equivalent) in caffe2/operators/hip as well

jerryzh168 commented:
@bddppq Yes I'm fixing them.

jerryzh168 committed:
Remaining Tensor API fixes(CPU) - dims() -> sizes()

Summary: Remaining fixes so that D12812029 will compile

Differential Revision: D13535559

fbshipit-source-id: a9f701028a42d641288fd6e30f12c42050069c27

jerryzh168 committed:
Remove deprecated caffe2::Tensor APIs (#15814)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15814

Plan is to remove the APIs we want to deprecate one by one and make sure it still builds in sandcastle and ossci

Reviewed By: ezyang

Differential Revision: D12812029

fbshipit-source-id: adc64d6b8373b21e70be017f06bf36e578e20a96
